viewing_party/party.py

```diff
@@ -1,23 +1,165 @@
+
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    watchlist = []
+    watchlist.append(movie)
+    user_data["watchlist"] = watchlist
+
+    return user_data
+
+def watch_movie(user_data, title):
+    # need to see if the title is in the usersdata
+    # if movie in WATCHLIST then remove the movie from watchlist 
+    # and add to watched, return
+    # if movie not in watchlist, return
+    for i in range(len(user_data['watchlist'])):
+        if user_data['watchlist'][i]['title'] == title:
+            user_data['watched'].append(user_data['watchlist'][i])
+            user_data['watchlist'].remove(user_data['watchlist'][i])
+            return user_data
+        
+    return user_data
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def average(lst):
+    return sum(lst) / len(lst)
 
+def get_watched_avg_rating(user_data):
+    rating_list = []
+    watched_list = user_data["watched"]
+    for movie in watched_list:
+        rating_list.append(movie['rating'])
+
+    if rating_list:
+        return average(rating_list)
+    else:
+        return 0.0
+
+def get_most_watched_genre(user_data):
+    most_watched_genre_dict = {}
+    watched_movies_list = user_data["watched"]
+    for movie in watched_movies_list:
+        genre = movie["genre"]
+        if genre not in most_watched_genre_dict:
+            most_watched_genre_dict[genre] = 1
+        else:
+            most_watched_genre_dict[genre] += 1
+
+    if most_watched_genre_dict:
+        max_val = list(most_watched_genre_dict.values())
+        max_key = list(most_watched_genre_dict.keys())
+        return max_key[max_val.index(max(max_val))]
+    else:
+        return None
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
+def get_unique_watched(user_data):
+    # create a list of movies watched by the user
+    # create a list of movies watched by the friends
+    # compare list of movies by user to movies by friend
+    # return list of movies only watched by user
+    # create a new unique list of movies watched by user
+    user_movie_list = user_watched_list(user_data)
+    friends_movie_list = friends_watched_list(user_data)
+    unique_movie_list = []
+
+    for movie in user_movie_list:
+        if movie not in friends_movie_list:
+            unique_movie_list.append(movie)
+    return unique_movie_list
+
+def get_friends_unique_watched(user_data):
+    user_movie_list = user_watched_list(user_data)
+    friends_movie_list = friends_watched_list(user_data)
+    friends_unique_movie_list = []
+
+    for movie in friends_movie_list:
+        if movie not in user_movie_list:
+            if movie not in friends_unique_movie_list:
+                friends_unique_movie_list.append(movie)
+    return friends_unique_movie_list
+    
+# HELPER FUNCTION
+def user_watched_list(user_data):

```

Comment:
Nice job creating helper functions! ✨ 

Consider if it's necessary to copy the movies from `user_data["watched"]` into a new list though. Now that we have talked about Big(O), we know that this adds a linear operation to both your time and space complexity. Since we already have access to a list of movies the user has watched in the provided data, couldn't we just use `user_data["watched"]` instead?